src/tools/wasm-emscripten-finalize.cpp

```diff
@@ -210,6 +210,7 @@ int main(int argc, const char* argv[]) {
 
   Module wasm;
   ModuleReader reader;
+  reader.setDebugInfo(debugInfo);
```

Comment:
Is this the right flag though?    What if I have a debug build that I want finalize to be fast on?   Wouldn't make more sense to set this to `!emitBinary` or maybe even better `!emitBinary && debugInfo`?

Reply:
This line just applies debugInfo from the commandline into the binary reader, which seems straightforward? If debug info handling is not requested, we don't even need to load it.

I agree this could perhaps be tied into `emitBinary` as well (if you are not writing anything, you are not writing debug info as well), but that does not improve things and adds a little more complexity to the code.

Reply:
But isn't the debug build case one of the main cases that we want to speed up.   In that case we are passing `-g` and we also don't want to modify the binary.   That is the case we care about the most isn't it?

Reply:
Yes, that is the case.

I am trying to keep this PR as simple as possible. The idea is when debug info is not used we don't need to load it - that seems straightforward?

There will be more complexity on the emscripten side, but we'll try to keep it simple there. But yes, it will consider whether it is writing a wasm from finalize when it decides to pass in `-g` or not.

Reply:
I think emscripten should still pass `-g` to signal that debug info should not be stripped.   

Whether we parse debug info here in binaryen should depend of whether we need to read/modify or write it, regardless of the `-g` flag.

Anyway, this change lgtm.

Reply:
I see your point of view as well. But consider the downside of this PR, which is that validation errors lose names (and have indexes instead). With this PR as it is written, you can re-add `-g` to get the name. If on the other hand we do it your way, then you'd need to add a `-o unused` for that. That seems less clear to me. But, I don't feel strongly.

Reply:
I don't think we should be depending on finalize to do validation.. its historically been a side effect, but we recently removed all code validation anyway.

If we want it to do validation (I'm not sure we will) then we can add an explicit `--validate` flag to disable both this and the code section skipping.

Reply:
By default we want finalize to be fast and skip any steps that it can which is why I think `--validate` should be opt in.

Reply:
(Obviously validation is required or some of things that finalize does so we have no choice in that case.. but when we do have a choice don't we want to skip by default?)

Reply:
> I don't think we should be depending on finalize to do validation.. its historically been a side effect, but we recently removed all code validation anyway.

Huh, that surprises me...

Indeed I don't see it in the code. But where did we do this? I don't see it in the `git log` for that file.

Reply:
Separately from that you are 100% right that this needs to check if we are writing output - the `lld` tests here fail on that. So that can't be done purely on the emscripten side. Anyhow, not a big deal, pushed to here.